nodes/modifier_make/contour2D.py

```diff
@@ -0,0 +1,599 @@
+# ##### BEGIN GPL LICENSE BLOCK #####
+#
+#  This program is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU General Public License
+#  as published by the Free Software Foundation; either version 2
+#  of the License, or (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, write to the Free Software Foundation,
+#  Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+#
+# ##### END GPL LICENSE BLOCK #####
+import itertools
+from collections import defaultdict
+from math import sin, cos, pi, degrees, radians, atan2, asin
+from mathutils.geometry import intersect_line_line as LineIntersect
+from mathutils import Vector
+
+import bpy
+import bmesh
+from bpy.props import BoolProperty, IntProperty, FloatProperty, EnumProperty
+
+from sverchok.node_tree import SverchCustomTreeNode
+from sverchok.data_structure import (
+    fullList, match_long_repeat,
+    dataCorrect, repeat_last,
+    updateNode, match_long_cycle)
+from sverchok.utils.sv_mesh_utils import mesh_join
+from sverchok.utils.cad_module_class import CAD_ops
+from sverchok.utils.sv_bmesh_utils import bmesh_from_pydata
+from sverchok.nodes.modifier_change.edges_intersect_mk2 import (
+    order_points,
+    remove_permutations_that_share_a_vertex,
+    get_valid_permutations,
+    can_skip,
+    get_intersection_dictionary,
+    update_mesh,
+    unselect_nonintersecting)
+
+
+modeItems = [
+    ("Constant", "Constant", ""),
+    ("Weighted", "Weighted", "")]
+
+listMatchItems = [
+    ("Long Repeat", "Long Repeat", ""),
+    ("Long Cycle", "Long Cycle", "")]
+
+
+def intersectEdges(verts, edges, epsi):
+    bm = bmesh_from_pydata(verts, edges, [])
+
+    edge_indices = [e.index for e in bm.edges]
+    trim_indices = len(edge_indices)
+    for edge in bm.edges:
+        edge.select = True
+
+    cm = CAD_ops(epsilon=epsi)
+    d = get_intersection_dictionary(cm, bm, edge_indices)
+    unselect_nonintersecting(bm, d.keys(), edge_indices)
+    # store non_intersecting edge sequencer
+    add_back = [[i.index for i in edge.verts] for edge in bm.edges if not edge.select]
+
+    update_mesh(bm, d)
+    verts_out = [v.co.to_tuple() for v in bm.verts]
+    edges_out = [[j.index for j in i.verts] for i in bm.edges]
+
+    # optional correction, remove originals, add back those that are not intersecting.
+    edges_out = edges_out[trim_indices:]
+    edges_out.extend(add_back)
+    bm.free()
+    return verts_out, edges_out
+
+
+def ptInTriang(p_test, p0, p1, p2):
+    # Function taken from Ramiro R.C https://stackoverflow.com/a/46409704
```

Comment:
consider adding this to https://github.com/nortikin/sverchok/blob/master/utils/modules/geom_utils.py , and importing it.